src/base/bittorrent/torrentimpl.cpp

```diff
@@ -1849,8 +1852,11 @@ void TorrentImpl::prepareResumeData(const lt::add_torrent_params &params)
     }
     else
     {
+        const auto preserveSeedMode = (!hasMetadata() && (m_ltAddTorrentParams.flags & lt::torrent_flags::seed_mode));
```

Comment:
I think `bool` isn't implied on the right hand side:
```suggestion
        const bool preserveSeedMode = (!hasMetadata() && (m_ltAddTorrentParams.flags & lt::torrent_flags::seed_mode));
```

Reply:
> I think `bool` isn't implied on the right hand side:

Result of `&&` is of type `bool`, isn't it?
But actually I just copy-paste it from other place where it's more explicit.

Reply:
Changed anyway.

Reply:
>Result of && is of type bool, isn't it?

If I'm not mistaken, operator `&&` can be overloaded so the return type isn't always `bool`.

Reply:
> > Result of && is of type bool, isn't it?
> 
> If I'm not mistaken, operator `&&` can be overloaded so the result type isn't always `bool`.

It is restricted to return `bool`.

Reply:
>https://en.cppreference.com/w/cpp/language/operator_logical

All built-in operators return bool, and most [user-defined overloads](https://en.cppreference.com/w/cpp/language/operators) also return bool so that the user-defined operators can be used in the same manner as the built-ins. **However, in a user-defined operator overload, any type can be used as return type (including void).**

Reply:
Ok, you're right. Documentation is slightly confusing since it has bool in method signatures whereas it would be more clear to have some type placeholder there.